subsys/bluetooth/host/conn_internal.h

```diff
@@ -62,6 +62,7 @@ enum {
 	BT_CONN_BR_NOBOND,                    /* SSP no bond pairing tracker */
 	BT_CONN_BR_PAIRING_INITIATOR,         /* local host starts authentication */
 	BT_CONN_CLEANUP,                      /* Disconnected, pending cleanup */
+	BT_CONN_AUTO_INIT_PROCEDURES_DONE,    /* Auto-initiated procedures have been done */
```

Comment:
I think it might be more intuitive if you reversed this, i.e. call it `AUTO_INIT_PROCEDURES_PENDING`. That way the reference is tied to something more tangible (the flag being set).

Reply:
I did this to distinguish between a freshly-memset connection and a connection that has had this procedure performed. I can swap if you really want to.

Reply:
I think verifying the correctness of the reference counting becomes clearer if it was reversed. Normally reference counts are tied to actual pointer variables, but that's not the case here. If you can see a setting of the flag when you do `ref()` and a `test_and_clear()` when you do `unref()` it's IMO more obvious to what the reference count is tied, i.e. something like:

```c
bt_conn_ref(conn);
set_bit();
```

and:

```c
if (test_and_clear_bit()) {
	bt_conn_unref();
}
```

Reply:
aiight, i'll change it.